pkg/crc/preflight/preflight_checks.go

```diff
@@ -0,0 +1,430 @@
+package preflight
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/user"
+	"regexp"
+	"strings"
+
+	units "github.com/docker/go-units"
+)
+
+const (
+	driverBinaryDir      = "/usr/local/bin"
+	kvmDriverBinaryPath  = driverBinaryDir + "/docker-machine-driver-kvm"
+	kvmDriverDownloadURL = "https://github.com/dhiltgen/docker-machine-kvm/releases/download/v0.10.0/docker-machine-driver-kvm-centos7"
+	defaultPoolSize      = "20 GB"
+	poolXMLDef           = `<pool type='dir'>
+	<name>default</name>
+		<target>
+			<path>/var/lib/libvirt/images</path>
+		</target>
+</pool>`
+
+	crcNetworkXMLDef = `<network>
+	<name>crc</name>
+	<uuid>49eee855-d342-46c3-9ed3-b8d1758814cd</uuid>
+	<forward mode='nat'>
+	  <nat>
+		<port start='1024' end='65535'/>
+	  </nat>
+	</forward>
+	<bridge name='tt0' stp='on' delay='0'/>
+	<mac address='52:54:00:fd:be:d0'/>
+	<domain name='test1.tt.testing' localOnly='yes'/>
+	<dns>
+	  <srv service='etcd-server-ssl' protocol='tcp' domain='test1.tt.testing' target='etcd-0.test1.tt.testing' port='2380' weight='10'/>
+	  <host ip='192.168.126.11'>
+		<hostname>api.test1.tt.testing</hostname>
+		<hostname>etcd-0.test1.tt.testing</hostname>
+	  </host>
+	</dns>
+	<ip family='ipv4' address='192.168.126.1' prefix='24'>
+	  <dhcp>
+		<host mac='52:fd:fc:07:21:82' name='test1-nfxdm-master-0' ip='192.168.126.11'/>
```

Comment:
This is risky... as the MAC is not static.
this needs to be part of a template perhaps...

Reply:
or this should be read from the archive with the disk images.

@praveenkumar how to handle a possible static mac?

Reply:
The mac address can be specified in the domain xml, and we can have a static mac address as constant, and use template to substitute it, in the network xml definition 

but the kvm machine driver's xml template doesn't have the field for it as of now.

Reply:
we should test if the mac can be changed from the image... as it is only the DNS overlay that might need it